applications/contrastive_phenotyping/evaluation/cosine_similarity.py

```diff
@@ -0,0 +1,369 @@
+                 color='blue', alpha=0.3, label='Std Dev (No Tracking)')
+
+plt.xlabel('Time Shift (τ)')
+plt.ylabel('Displacement')
+plt.title('Embedding Displacement Over Time')
+plt.grid(True)
+plt.legend()
+
+plt.savefig('embedding_displacement_euclidean.svg', format='svg')
+plt.savefig('embedding_displacement_euclidean.pdf', format='pdf')
+
+# Show the Euclidean plot
+plt.show()
+
+# %% Plot 2: Cosine Displacements
+plt.figure(figsize=(10, 6))
+
+# Plotting Cosine displacements
+mean_values_30_min_cosine = list(mean_displacement_30_min_cosine.values())
+std_values_30_min_cosine = list(std_displacement_30_min_cosine.values())
+
+mean_values_no_track_cosine = list(mean_displacement_no_track_cosine.values())
+std_values_no_track_cosine = list(std_displacement_no_track_cosine.values())
+
+plt.plot(taus, mean_values_30_min_cosine, marker='o', label='Cell & Time Aware (30 min interval)')
+plt.fill_between(taus, np.array(mean_values_30_min_cosine) - np.array(std_values_30_min_cosine), np.array(mean_values_30_min_cosine) + np.array(std_values_30_min_cosine), 
+                 color='orange', alpha=0.3, label='Std Dev (30 min interval, Cosine)')
+
+plt.plot(taus, mean_values_no_track_cosine, marker='o', label='Classical Contrastive (No Tracking)')
+plt.fill_between(taus, np.array(mean_values_no_track_cosine) - np.array(std_values_no_track_cosine), np.array(mean_values_no_track_cosine) + np.array(std_values_no_track_cosine), 
+                 color='red', alpha=0.3, label='Std Dev (No Tracking)')
+
+plt.xlabel('Time Shift (τ)')
+plt.ylabel('Cosine Dissimilarity')
+plt.title('Embedding Displacement Over Time')
+plt.grid(True)
+plt.legend()
+
+# Show the Cosine plot
+plt.show()
+# %%
+
+import seaborn as sns
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from pathlib import Path
+from collections import defaultdict
+from sklearn.metrics.pairwise import cosine_similarity
+from viscy.representation.embedding_writer import read_embedding_dataset
+
+# %% Paths to datasets
+features_path_30_min = Path("/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/time_sampling_strategies/time_interval/predict/feb_test_time_interval_1_epoch_178.zarr")
+feature_path_no_track = Path("/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/time_sampling_strategies/negpair_random_sampling2/feb_fixed_test_predict.zarr")
+
+# %% Read embedding datasets
+embedding_dataset_30_min = read_embedding_dataset(features_path_30_min)
+embedding_dataset_no_track = read_embedding_dataset(feature_path_no_track)
+
+# %% Function to compute the norm of differences between embeddings at t and t + tau
+# %% Function to compute the norm of differences between embeddings at t and t + tau
+def compute_displacement(embedding_dataset, max_tau=10, use_cosine=True):
```

Comment:
@alishbaimran this should be part of `viscy.representation.evaluation`. All algorithms should be there. The scripts should have only paths and plotting code.